pkg/prom/query.go

```diff
@@ -157,46 +166,68 @@ func runQuery(query string, ctx *Context, resCh QueryResultsChan, profileLabel s
 	resCh <- results
 }
 
-func (ctx *Context) query(query string) (interface{}, prometheus.Warnings, error) {
+// RawQuery is a direct query to the prometheus client and returns the body of the response
+func (ctx *Context) RawQuery(query string) ([]byte, error) {
 	u := ctx.Client.URL(epQuery, nil)
 	q := u.Query()
 	q.Set("query", query)
 	u.RawQuery = q.Encode()
 
 	req, err := http.NewRequest(http.MethodPost, u.String(), nil)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
-	resp, body, warnings, err := ctx.Client.Do(context.Background(), req)
-	for _, w := range warnings {
-		// NoStoreAPIWarning is a warning that we would consider an error. It returns partial data relating only to the
-		// store apis which were reachable. In order to ensure integrity of data across all clusters, we'll need to identify
-		// this warning and convert it to an error.
-		if IsNoStoreAPIWarning(w) {
-			return nil, warnings, NewCommError(fmt.Sprintf("Error: %s, Body: %s, Query: %s", w, body, query))
-		}
-
-		log.Warningf("fetching query '%s': %s", query, w)
+	// Set QueryContext name if non empty
+	if ctx.name != "" {
+		req = httputil.SetName(req, ctx.name)
 	}
+	req = httputil.SetQuery(req, query)
+
+	// Note that the warnings return value from client.Do() is always nil using this
+	// version of the prometheus client library. We parse the warnings out of the response
+	// body after json decodidng completes.
```

Comment:
Hopefully this comment sheds light on the current scenario. The prometheus Warnings are actually part of the response payload. Since the http request will never return warnings, we needed to implement our own warning parsing, which happens after the unmarshalling.